apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx

```diff
@@ -58,6 +73,36 @@ const PreviewPreEdit = ({ navigation, route }: NavigationProps) => {
     [navigation, device],
   );
 
+  const nftMetadata = useNftMetadata(
+    nft?.contract,
+    nft?.tokenId,
+    nft?.currencyId,
+  );
+
+  const { status, metadata } = nftMetadata as NFTResource & {
+    metadata: NFTMetadata;
+  };
+
+  const nftMediaTypes = useMemo(
+    () => (metadata ? getMetadataMediaTypes(metadata) : null),
+    [metadata],
+  );
+
+  const nftMediaSize = useMemo(
+    () =>
+      nftMediaTypes
+        ? (["big", "preview"] as NFTMediaSize[]).find(
+            size => nftMediaTypes[size] === "image",
+          )
+        : null,
+    [nftMediaTypes],
+  );
+
+  const nftImageUri = useMemo(
+    () => (nftMediaSize && metadata?.medias?.[nftMediaSize]?.uri) || null,
+    [nftMediaSize, metadata],
+  );
+
```

Comment:
can we maybe extract that into a single hook? like `useNftCustomImageUri`

Reply:
Done with what we talked about on slack